dask/dataframe/tests/test_dataframe.py

```diff
@@ -4135,6 +4135,24 @@ def test_set_index_with_index():
     assert ddf2 is ddf
 
 
+def test_set_index_no_sort():
+    """
+    GH10333 - Allow setting index on existing partitions without
+    computing new divisions and repartitioning.
+    """
+    df = dd.from_dict({"col1": [2, 4, 1, 3, 5], "col2": [1, 2, 3, 4, 5]}, npartitions=2)
+
+    # Default is sort=True
+    result = df.set_index("col1")
+    assert result.divisions == (1, 2, 5)
```

Comment:
Actually I wonder if we already have coverage for the default in other tests. We may be able to just test the new `sort=False` case here

Reply:
Good tip! I looked around a bit at current tests having `set_index` and from what I see, they were testing fairly specific behaviors and thought conflating them to include this wasn't worth it; but happy to take suggestions if you see otherwise.
https://github.com/dask/dask/pull/10342/commits/4a5e319c66789c40c33bc7d0f2d6daa569a679c5